core/server/api_container/server/startosis_engine/startosis_interpreter.go

```diff
@@ -124,16 +122,21 @@ func (interpreter *StartosisInterpreter) Interpret(
 	var argsTuple starlark.Tuple
 	var kwArgs []starlark.Tuple
 
-	if shouldInjectPlanArg || (isUsingDefaultMainFunction && mainFunction.NumParams() >= minimumParamsRequiredForPlan) {
-		if paramName, _ := mainFunction.Param(planParamIndex); paramName != planParamName {
-			return "", nil, startosis_errors.NewInterpretationError(unexpectedArgNameError, planParamIndex, planParamName, paramName).ToAPIType()
+	mainFuncParamsNum := mainFunction.NumParams()
+
+	if mainFuncParamsNum >= minimumParamsRequiredForPlan {
+		// the plan object will always be injected if the first argument name is 'plan'
+		firstParamName, _ := mainFunction.Param(planParamIndex)
```

Comment:
cool the plan param can only be the first param?

Reply:
can we for loop a non run functions params and see which one is plan?

Reply:
yes, I think we could do that, I'll do it on the next PR. I want to have this merged